src/app/modules/shared_models/social_links_model.nim

```diff
@@ -0,0 +1,114 @@
+import NimQml, tables, sequtils, sugar
+
+import ../../../app_service/common/social_links
+
+import social_link_item
+
+proc toSocialLinkItems*(source: SocialLinks): seq[SocialLinkItem] =
+  proc textToType(text: string): LinkType =
+    if (text == SOCIAL_LINK_TWITTER_ID): return LinkType.Twitter
+    if (text == SOCIAL_LINK_PERSONAL_SITE_ID): return LinkType.PersonalSite
+    if (text == SOCIAL_LINK_GITHUB_ID): return LinkType.Github
+    if (text == SOCIAL_LINK_YOUTUBE_ID): return LinkType.Youtbue
+    if (text == SOCIAL_LINK_DISCORD_ID): return LinkType.Discord
+    if (text == SOCIAL_LINK_TELEGRAM_ID): return LinkType.Telegram
+    return LinkType.Custom
+  result = map(source, x => initSocialLinkItem(x.text, x.url, textToType(x.text)))
+
+type
+  ModelRole {.pure.} = enum
+    Uuid = UserRole + 1
+    Text
+    Url
+    LinkType
+
+QtObject:
+  type
+    SocialLinksModel* = ref object of QAbstractListModel
+      items: seq[SocialLinkItem]
+
+  proc delete(self: SocialLinksModel) =
+    self.items = @[]
+    self.QAbstractListModel.delete
+
+  proc setup(self: SocialLinksModel) =
+    self.QAbstractListModel.setup
+
+  proc newSocialLinksModel*(): SocialLinksModel =
+    new(result, delete)
+    result.setup
+
+  proc setItems*(self: SocialLinksModel, items: seq[SocialLinkItem]) =
+    self.beginResetModel()
+    self.items = items
+    self.endResetModel()
+
+  proc appendItem*(self: SocialLinksModel, item: SocialLinkItem) =
+    self.beginInsertRows(newQModelIndex(), self.items.len, self.items.len)
+    self.items.add(item)
+    self.endInsertRows()
+
+  proc removeItem*(self: SocialLinksModel, uuid: string): bool =
+    for i in 0 ..< self.items.len:
+      if (self.items[i].uuid == uuid):
+        if (self.items[i].linkType == LinkType.Custom):
+          self.beginRemoveRows(newQModelIndex(), i, i)
+          self.items.delete(i)
+          self.endRemoveRows()
+          return true
+        return false
+    return false
+
+  proc updateItem*(self: SocialLinksModel, uuid, text, url: string): bool =
```

Comment:
What about doing it the right way (tm) by overriding the setData() method? :)

See https://doc.qt.io/qt-5/qabstractitemmodel.html#setData